### ISSUE TITLE
FX2D display is inverted in 3.0b6

### ISSUE THREAD

**Issue by monkstone** [processing/processing#3795]
![oops](https://cloud.githubusercontent.com/assets/86850/9835454/26e5decc-59e4-11e5-9a19-15efbc9bb0ea.png)

**Comment by StanLepunK** [comment]
same on OSX 10.9.5

````
void setup() {
  size(300,300,FX2D) ;
}
void draw() {
  background(0) ;
  text("test",80,80) ;
}
````

**Comment by jacbouzada** [comment]
Same problem in OS X 10.10.5

**Comment by adegani** [comment]
Same problem (Linux). Processing-3.0b5 was OK!

**Comment by brendanmatkin** [comment]
confirm 10.10.5

**Comment by JakubValtar** [comment]
For now, use `noSmooth()` in your `setup()` as a workaround. It looks like FX has buggy antialiasing.

**Comment by benfry** [comment]
Found a workaround for 3.0 beta 7.

**Comment by benfry** [comment]
Nuts... maybe not. Think I was fooled by this retina display.

**Comment by benfry** [comment]
So we can do one of the following: 

1) revert to the previous Java release before 8u60 (inconvenient for anyone building the source since they'll need to find/install an older Java release to build)
2) leave this broken until it's fixed upstream in Java (but it's a nasty, obvious bug)
3) check the Java version and do an axis flip if on OS X and Linux (need to make sure this would work and not introduce weirder off-by-1-pixel scenarios) 
4) disable smoothing when using 8u60 and OS X + Linux

Your vote, @JakubValtar?

**Comment by JakubValtar** [comment]
I know you secretly wish we could get away with 2), I also do, but we are too nice for that. I would otherwise vote for 3), but I'm afraid it will be messy and people will break it (they always do). So, my proposal is:

We want most people to have antialiasing, so let's bundle 8u51 with the release so there is no compromise for majority of end users.

It also has to run on all recent JREs, so let's also disable smoothing for 8u60 + OS X & Linux combos and print warning that FX in 8u60 does not smooth if people try to enable it. And it's just a few lines of code so there is not much we can break. This covers developers & people who like to run sketches with different JREs (i.e. me & me). These people have enough knowledge to install 8u51 or switch renderer if they can't live without smoothing.

**Comment by benfry** [comment]
2 is the worst of the bunch, I think the third is cleanest/best if it actually works, followed by the fourth. Or maybe 4 and 1.

**Comment by JakubValtar** [comment]
As I said, 3 will be messy and hard to get right and 4 alone has too broad impact. So I'm for 4+1 since 1 is really easy and will give smoothing to the masses and 4 covers devs.

**Comment by jacbouzada** [comment]
One vote for the 3rd option ;)  
(but I don't know how messy/hard it would be....)

**Comment by benfry** [comment]
3 is too messy and will introduce more bugs than it fixes.

Going with 4 and 1 for the next release.

**Comment by tillnagel** [comment]
Just to let you know, I updated the Wiki page on how to build Processing to reflect this latest change. https://github.com/processing/processing/wiki/Build-Instructions

**Comment by monkstone** [comment]
Hi just to note @benfry problem does not seem to be resolved with recently released jdk1.8.0_66-b17

**Comment by benfry** [comment]
Yep, I don't expect it until 8u80.

**Comment by benfry** [comment]
Fixed in 8u72, and Processing 3.0.2 will ship with 8u74 so we're all set on this one.
